### PR TITLE
EN-59946: update Jenkinsfile

### PR DESCRIPTION
**Pull request by sarah-salinas**
## Before

In the Jenkinsfile, on release cuts, the dockerized image is deployed to RC.

## After

In the Jenkinsfile, on release builds, the dockerized image is not deployed to RC.
Also:
- new parameter RELEASE_DRY_RUN added to ease testing
- changed stage name from "Cut" to "Release Tag"
 - used simpler syntax for deploying
 - simplified logic for which stage to build

### CODE REVIEW

**Comment by sarah-salinas** [comment]
## Testing

- PR Checks: https://jenkins-build.socrata.com/job/data-platform/job/soql-server/job/Soql-server-pg%20Build%20Pipeline/job/PR-541/6/
  - stage: Build Server
  - build description:  soql-server-pg:5.1.21-SNAPSHOT_6_7a48cc74 - SHA matches the most recent commit on the PR branch
- Simulated main branch build (on my branch): https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/140/console
  - stages: Build Server, Dockerize Server, Dockerize Secondary, Deploy, Deploy PG Control Mirrors, Deploy Citus Mirrors
  - build description / deploy tag: soql-server-pg:5.1.21-SNAPSHOT_140_7a48cc74 - SHA matches most recent commit on my branch
- Release build (dry run): https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/141/console
  - stages: Release Tag, Build Server, Dockerize Server, Dockerize Secondary
  - build description / deploy tag: soql-server-pg:5.1.20_141_e95dc926 - SHA matches last release tag

**Comment by sarah-salinas** [comment]
## Testing again

- Simulated main branch build (on my branch): https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/143/console
- Release build (dry run): https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/142/console

File: Jenkinsfile [hunk 1]
```diff
@@ -2,223 +2,172 @@
...
+            echo 'DRY RUN: Skipping release tag creation'
           }
           else {
-            echo 'Running sbt-release'
+            // get a list of all files changes since the last tag
+            files = sh(returnStdout: true, script: "git diff --name-only HEAD `git describe --match \"v*\" --abbrev=0`").trim()
+            echo "Files changed:\n${files}"
 
-            // The git config setup required for your project prior to running 'sbt release with-defaults' may vary:
-            sh(returnStdout: true, script: "git config user.name \'Jenkins Server in aws-us-west-2-infrastructure\'")
```
Comment (sarah-salinas) [inline Jenkinsfile:118]: I don't think we need lines 118-120 since they are not in comparable jobs.

**Comment by sarah-salinas** [comment]
Secondary deploys were taking more than an hour . . . I believe that the service version variable was misconfigured.  I've fixed it and will test again:

Simulated main build (on my branch)
https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/144/console

**Comment by sarah-salinas** [comment]
## Testing again

- PR Check
  - stage: Build SoQL Server PG
  - Description: soql-server-pg:5.1.21-SNAPSHOT_12_03f8f644 

- Simulated Main branch build (on this branch) - https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/146/consoleFull
  - stages: Build SoQL Server PG, Build SoQL Server PG, Dockerize Secondary Watcher, Deploys:, Deploy SoQL Server PG, Deploy Secondary Watcher PG, Deploy SoQL Server PG Control Mirror, Deploy Secondary Watcher PG Control Mirror, Deploy SoQL Server Citus Mirror', Deploy Secondary Watcher Citus Mirror
  - description / deploy tag: soql-server-pg:5.1.21-SNAPSHOT_146_03f8f644

- Release build (dry run): https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/147/console
  - stages: Release Tag, Build SoQL Server PG, Build SoQL Server PG, Dockerize Secondary Watcher
  - description / deploy tag: soql-server-pg:5.1.20_147_e95dc926

**Comment by sarah-salinas** [comment]
## Testing again

I made some changes to the deploy stages and so I've tested again.

- PR checks ✅ 
- simulated main: https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/148/console ✅ 
- release dry run: https://jenkins-build.socrata.com/view/Cut%20Jobs/job/soql-postgres-adapter-cut/149/console ✅

File: Jenkinsfile [hunk 2]
```diff
@@ -2,223 +2,174 @@
...
-          deploy.checkoutAndInstall()
-
-          deploy.deploy("soql-server-mirror-citus1-staging", deploy_environment, dockerize_server.getDeployTag())
-          deploy.deploy("secondary-watcher-mirror-citus*", deploy_environment, dockerize_secondary.getDeployTag())
+      stages {
+        stage('Deploy SoQL Server PG') {
+          steps {
+            script {
+              // uses env.DOCKER_TAG and deploys to staging by default
+              marathonDeploy(serviceName: env.DEPLOY_PATTERN, waitTime: '60')
```
Comment (sarah-salinas) [inline Jenkinsfile:140]: soql-server-pg* needs a high wait time for the deploy - https://jenkins-build.socrata.com/job/soql-postgres-adapter-cut/109/flowGraphTable/ - this is an example where it took 48 minutes to deploy soql-server-pg* to rc.

File: Jenkinsfile [hunk 3]
```diff
@@ -2,223 +2,174 @@
...
+            if (params.RELEASE_BUILD){
+              echo env.SECONDARY_DOCKER_TAG // For now, just print the deploy tag in the console output -- later, communicate to release metadata service
+            }
+          }
         }
-
-    stage('Deploy Citus Mirrors') {
+      }
+    }
+    stage('Deploys:') {
```
Comment (sarah-salinas) [inline Jenkinsfile:128]: I've added an outer layer stage so that the `when` condition will apply to all deploy stages.

**Review by saraid — APPROVED** [review]
I'd prefer it if @zachlefevre took a look, but I'm not finding a problem.

**Comment by zachlefevre** [comment]
Looks reasonable!

**Review by zachlefevre — APPROVED** [review]

**Pull request merged**